app/src/pages/authoring/[resourceType]/[id].tsx

```diff
@@ -86,18 +132,87 @@ export default function ResourceAuthoringPage() {
       <Grid>
         <Grid.Col span={6}>
           <Stack spacing="md">
-            <TextInput label="url" value={url} onChange={e => setUrl(e.target.value)} />
-            <TextInput label="identifier" value={identifier} onChange={e => setIdentifier(e.target.value)} />
+            <TextInput
+              label="url"
+              value={url}
+              onChange={e => setUrl(e.target.value)}
+              rightSection={
+                <ActionIcon
+                  onClick={() => {
+                    setUrl('');
+                  }}
+                >
+                  <X />
+                </ActionIcon>
+              }
+            />
```

Comment:
Since you have a lot of TextInput elements and their layout seems to be the same, would it be helpful to pull this TextInput div into another file and then call it as many times as you need it and pass in the properties as params? This might result in a trade-off I'm not aware of, however.

Reply:
Great suggestion! I was able to pull this into a file in the components directory called `ArtifactFieldInput.tsx`